### DISCUSSION TITLE
Should scripts work with .scala files

### DISCUSSION THREAD

**Discussion by alvinj** [VirtusLab/scala-cli#442] [Using directives and cmd configuration options]
I was just wondering if `.sc` scripts are supposed to work with `.scala` files that are in the current directory, or in a resources directory?

I tried this, with both files in the same directory, and it didn’t work:

```scala
// Person.scala
case class Person(name: String)

// Test.sc
#!/usr/bin/env scala-cli
val p = Person("Jane")
println(p.name)
```

Then I tried this and it also didn’t work:

```scala
// resources/Person.scala
case class Person(name: String)

// Test.sc
#!/usr/bin/env scala-cli
// using resourceDir "./resources"
val p = Person("Jane")
println(p.name)
```

That second attempt results in this error:

```sh
$ ./TestPerson.sc 
[error] ./TestPerson.sc:2:4: Unrecognized directive: using resourceDir ./resources
[error] // using resourceDir "./resources"
[error]    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

This isn’t a huge deal, just a question. I didn’t know if scripts were meant to be totally included in one file, but then I saw that they work with jar files, so it made me wonder what the difference was between using a `.scala` file in the current directory (or a resources directory) compared to using a jar file. Thanks.

Totally unrelated, but it’s cool that `using` directives like these work:

```scala
// using options -deprecation -unchecked -feature -explain
// using jar stringutils_3-0.1.jar
// using javaOpt -Xms64M -Xmx128M
```

Once you get used to it, it’s a really clever idea.

**Comment by tpasternak** [comment]
Yep, thank you for pointing it out! We need to somehow document/clarify this.

The first example would work if you run `scala-cli .` or `scala-cli Test.scala Person.scala`. If you point to a single file, the other ones are not compiled

`using resourceDir` was not found because we have had documentation mismatch (same problem as here https://github.com/VirtusLab/scala-cli/issues/432). In 0.0.9 it should be OK

**Comment by SethTisue** [comment]
Note that I also opened #466 which suggests that scripts with no file extension at all ought to be allowed.
